apps/system-e2e/src/support/wire-mocks.ts

```diff
@@ -149,16 +147,20 @@ export const addXroadMock = async <Conf extends XroadSectionConfig>(
   const stubResponses = Array.isArray(options.response)
     ? options.response
     : [options.response]
+
+  const [apiPath, query] = options.apiPath.split('?')
+  const queries = query
+    ? Array.from(new URLSearchParams(query)).map(([key, value]) => ({
+        [key]: value,
+      }))
+    : undefined
+
   const stub = new Stub().withPredicate(
-    options.query
-      ? new FlexiPredicate()
-          .withOperator(Operator.deepEquals)
-          .withPath(`${prefix}${env}${path}${options.apiPath}`)
-          .withMethod(method)
-          .withQuery(options.query)
-      : new EqualPredicate()
-          .withPath(`${prefix}${env}${path}${options.apiPath}`)
-          .withMethod(method),
+    new FlexiPredicate()
+      .withOperator(Operator.deepEquals)
+      .withPath(`${prefix}${env}${path}${apiPath}`)
+      .withMethod(method)
+      .withQuery(queries),
```

Comment:
Won't `.withQuery` fail if `queries` is `undefined`?

Reply:
Seems to get ignored if undefined